group/views.py

```diff
@@ -38,10 +41,7 @@ def get_group(group_id):
         group = Group.objects.get(id = group_id)
     except Group.DoesNotExist:
         logger.warning('Group: Can not edit group %s! Group is not exist!' % group_id)
-        return render(
-            request,
-            'index/404.html',
-            {'error_message': 'Group: Can not edit group %s! Group is not exist!' % group_id})
+        raise Http404('Group: Can not edit group %s! Group is not exist!' % group_id)
     return group
 
 def get_running_contest(request, group_id):
```

Comment:
Maybe you can use a filter to replace line 52~57
since ``group.trace_contest.all()`` is a ``QuerySet``.